ui/src/components/query/builder/edit/dialogs/SaveCustomSetDialog.vue

```
@@ -0,0 +1,229 @@
+
+const schemeOptions: Ref<TTIriRef[]> = ref([]);
+const typeOptions: Ref<TTIriRef[]> = ref([]);
+const selectedMember: Ref<Node> = ref({});
+const showSaveCustomSetDialog = ref(false);
+
+watch(
+  () => showSaveCustomSetDialog.value,
+  async () => await init()
+);
+
+onMounted(async () => {
+  await init();
+});
+
+async function init() {
+  typeOptions.value = await getTypeOptions();
+  schemeOptions.value = await getSchemeOptions();
+  selectDefaults();
+}
+
+async function isValidIri() {
+  if (!scheme.value.modelValue) return false;
+  const exists = await EntityService.iriExists(fullIri.value);
+  return !exists;
+}
+
+async function getTypeOptions(): Promise<TTIriRef[]> {
+  const setTypes = await EntityService.getEntityChildren(IM.SET);
+  return setTypes.map(setType => {
+    return {
+      "@id": setType["@id"],
+      name: setType.name
+    };
+  });
+}
+
+async function getSchemeOptions(): Promise<TTIriRef[]> {
+  const schemes = await EntityService.getEntityChildren(IM.NAMESPACE + "Graph");
+  return schemes.map(scheme => {
+    return {
+      "@id": scheme["@id"],
+      name: scheme.name
+    };
+  });
+}
+
+function onNameGenIri() {
+  setFieldValue("iri", name.value.modelValue);
+}
+
+function selectDefaults() {
+  setFieldValue("scheme", IM.NAMESPACE);
+  setFieldValue("type", IM.CONCEPT_SET);
+}
+
+const onSubmit = handleSubmit(async () => {
+  const setEntity = buildSetEntity();
+  try {
+    await FilerService.fileEntity(setEntity, IM.NAMESPACE, IM.ADD_QUADS);
+    const createdEntity = await EntityService.getFullEntity(setEntity["@id"]);
+    if (isObjectHasKeys(createdEntity, [RDFS.LABEL, RDF.TYPE, IM.HAS_STATUS, IM.SCHEME, IM.IS_CONTAINED_IN, IM.DEFINITION]))
+      toast.add({ severity: "success", summary: "Created", detail: 'Created "' + createdEntity[RDFS.LABEL], life: 3000 });
+  } catch (e: any) {
+    toast.add({ severity: "error", summary: e.response.data.title, detail: e.response.data.detail, life: 3000 });
```

why catch error and not let the error propagate through the error handler?

---

That would go to a 500 page losing all progress in the query builder. Better to catch the error here and not let the app fail.